### PR TITLE
[BZ2038166] Add ShiftStack legacy certs script

### PR DESCRIPTION
**Pull request by maxwelldb**
Context: https://bugzilla.redhat.com/show_bug.cgi?id=2038166

Preview 1: https://deploy-preview-40785--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/preparing-to-install-on-openstack.html#security-osp-validating-certificates_preparing-to-install-on-openstack

### CODE REVIEW

**Comment by maxwelldb** [comment]
@pierreprinetti FYI

This PR just has a new module at the moment--I haven't placed it in any docs assemblies, yet.

**Comment by maxwelldb** [comment]
TODO: Create 4.10 targeted PR for RN update

File: modules/security-osp-validating-certificates.adoc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+** grep
+** link:https://docs.openstack.org/python-openstackclient/latest/[openstack-cli]
+** link:https://stedolan.github.io/jq/[jq]
+** link:https://www.openssl.org/[openssl version 1.1.1l or greater]
+* Populate the machine with {rh-openstack} credentials for the target cloud.
+
+
+.Procedure
+
+. Save the following script to your machine:
```
Comment (maxwelldb) [inline modules/security-osp-validating-certificates.adoc:25]:
```suggestion
. Save the following script to your machine: 
+
```

**Comment by maxwelldb** [comment]
Added mod to one existing assembly. I need to create a new OSP-specific update assembly that will contain solely this, too.

**Comment by pierreprinetti** [comment]
This PR changes the script in the OpenShift upstream docs (it had syntax issues): https://github.com/openshift/installer/pull/5576

**Comment by maxwelldb** [comment]
@pierreprinetti 4e0d4ce adds your changes from openshift/installer#5576

**Review by pierreprinetti — COMMENTED** [review]
I have left a couple comments. One more thing that may or may not be interesting, is that users may look up the literal error that Go (and therefore operators) may return in the logs (see https://github.com/openshift/installer/blob/master/docs/user/openstack/known-issues.md#https-commonname-deprecation). You may or may not want to add it somewhere to help searches land on this page

Note that the error is returned if the user didn't prepare and actually upgrade while still using invalid certs.

File: modules/security-osp-validating-certificates.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
+// This is included in the following assemblies:
+//
+// * installing/installing_openstack/preparing-to-install-on-openstack.adoc
+// TODO:
+// * Add statement to all installation assembly prereqs
+// * Create new OSP update assembly
+[id="security-osp-validating-certificates_{context}"]
+= Scanning {rh-openstack} endpoints for legacy HTTPS certificates
+
+Beginning with {product-title} 4.10, HTTPS certificates must contain subject alternative name (SAN) fields. Run the following script to scan each HTTPS endpoint in a {rh-openstack-first} catalog for legacy certificates that do not contain these fields.
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply modules/security-osp-validating-certificates.adoc:10]: Works for me. Thanks!

```suggestion
Beginning with {product-title} 4.10, HTTPS certificates must contain subject alternative name (SAN) fields. Run the following script to scan each HTTPS endpoint in a {rh-openstack-first} catalog for legacy certificates that only contain the `CommonName` field.
```

**Comment by maxwelldb** [comment]
> I have left a couple comments. One more thing that may or may not be interesting, is that users may look up the literal error that Go (and therefore operators) may return in the logs (see https://github.com/openshift/installer/blob/master/docs/user/openstack/known-issues.md#https-commonname-deprecation). You may or may not want to add it somewhere to help searches land on this page
> 
> Note that the error is returned if the user didn't prepare and actually upgrade while still using invalid certs.

I think this is a good idea. Will do.

**Comment by maxwelldb** [comment]
@itzikb-redhat Can you QE this re: BZ2038166?

**Comment by itzikb-redhat** [comment]
LGTM

File: modules/security-osp-validating-certificates.adoc [hunk 3]
```diff
@@ -0,0 +1,114 @@
+// This is included in the following assemblies:
+//
+// * installing/installing_openstack/preparing-to-install-on-openstack.adoc
+// TODO:
+// * Add statement to all installation assembly prereqs
+// * Create new OSP update assembly
```
Comment (maxwelldb) [inline modules/security-osp-validating-certificates.adoc:6]:
```suggestion

:_content-type: ASSEMBLY
```

File: modules/security-osp-validating-certificates.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+Beginning with {product-title} 4.10, HTTPS certificates must contain subject alternative name (SAN) fields. Run the following script to scan each HTTPS endpoint in a {rh-openstack-first} catalog for legacy certificates that only contain the `CommonName` field.
+
+[IMPORTANT]
+{product-title} does not check the underlying {rh-openstack} infrastructure for legacy certificates prior to installation or updates. Use the provided script to check for these certificates yourself. Failing to update legacy certificates prior to installing or updating a cluster will result in cluster dysfunction.
+
+.Prerequisites
+
+* On the machine where you run the script, have the following software:
+** Bash version 4.0 or greater
+** grep
```
Comment (adellape) [inline modules/security-osp-validating-certificates.adoc:18]: Maybe markup `grep` tho I don't see it uniformly done in the repo so far. :shrug:

File: modules/security-osp-validating-certificates.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+** link:https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.2/html/command_line_interface_reference/the_openstack_client[OpenStack client]
+** link:https://stedolan.github.io/jq/[jq]
```
Comment (adellape) [inline modules/security-osp-validating-certificates.adoc:20]: Same thought for `jq` re: markup.

File: modules/security-osp-validating-certificates.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+// * installing/installing_openstack/preparing-to-install-on-openstack.adoc
+
+:_content-type: ASSEMBLY
+[id="security-osp-validating-certificates_{context}"]
+= Scanning {rh-openstack} endpoints for legacy HTTPS certificates
+
+Beginning with {product-title} 4.10, HTTPS certificates must contain subject alternative name (SAN) fields. Run the following script to scan each HTTPS endpoint in a {rh-openstack-first} catalog for legacy certificates that only contain the `CommonName` field.
+
+[IMPORTANT]
+{product-title} does not check the underlying {rh-openstack} infrastructure for legacy certificates prior to installation or updates. Use the provided script to check for these certificates yourself. Failing to update legacy certificates prior to installing or updating a cluster will result in cluster dysfunction.
```
Comment (adellape) [inline modules/security-osp-validating-certificates.adoc:12]: Ooh "cluster dysfunction", I like that.

File: modules/security-osp-validating-certificates.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+** link:https://stedolan.github.io/jq/[jq]
+** link:https://www.openssl.org/[OpenSSL version 1.1.1l or greater]
+* Populate the machine with {rh-openstack} credentials for the target cloud.
+
+
+.Procedure
+
+. Save the following script to your machine: 
++
+[source,bash]
```
Comment (adellape) [inline modules/security-osp-validating-certificates.adoc:29]: Maybe a `[%collapsible]` around this block?

File: modules/security-osp-validating-certificates.adoc (hunk 4 above)
Comment (maxwelldb) [reply modules/security-osp-validating-certificates.adoc:18]:
```suggestion
** `grep`
```

File: modules/security-osp-validating-certificates.adoc (hunk 5 above)
Comment (maxwelldb) [reply modules/security-osp-validating-certificates.adoc:20]:
```suggestion
** link:https://stedolan.github.io/jq/[`jq`]
```

**Pull request merged**